### ISSUE TITLE
Unexpected type errors with `any` and `unknown` combined with `not` pattern

### ISSUE THREAD

**Issue by m-rutter** [gvergnaud/ts-pattern#43]
**Describe the bug**
I had a type like this:

```ts
type Foo = { date?: any };
```

I meant for this to be a `{date?: Date}`, but nevermind - blame codegen.

I tried to match on it with `not(__.nullish)`, and I got some unsual errors:

```ts
match(foo).with({date: not(__.nullish)}, (foo) => {
    const date = foo.date;
})

// Property 'date' does not exist on type 'SeveralAnonymousSelectError<"You can only use a single anonymous selection (with `select()`) in your pattern. If you need to select multiple values, give them names with `select(<name>)` instead">'.(2339)
``` 

It seems trying to use `not` with any `any` or `unknown` values with any type of pattern produces this error. Now it doesn't make any sense to use `not` on `any` or `unknown`, but these errrors are quite confusing and appear unrelated to the mistake I've made. I haven't looked at it closely, but I'm guessing `SeveralAnonymousSelectError` is some kind of fallthrough case that this pattern is unintentionally hitting. 

My expectation would be that either we would get a more relevant error (something about how negated types aren't possible or something), or `date` would remian `any`.

I'll look into it more closely myself later.

**Code Sandbox with a minimal reproduction case**
[playground example](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbxAQxgYwBYBo4DsIw4D6RAvnAGZQQhwDkMAzgLRiowCmUudA3AFD8AJhzQAbZFA5w0EXI3gUIEAFyIhqDgH41yXAE9SAkeMnTZ8+ACNkALzUINnHXACuuANb4A7riOD+FHQMAAolCABKADpvYBhQx001fBgQkijcVzExYEYMCNIcMOUIuABeAD5EfjhamTkFOCdpMsplKOaBUgjBIMxiyJi4hObkghD3EQpgXA4hAqLw0srquvrLJs1ytogOzS6ewNR+paH4kMTOMdTM7IW4AeWqhBq6i0bm7fC9zgPe49Cp1i50uHDU6QUUBmAHNCg8luVnq9au94J9Wt9Ovxuv9giEbLZosCRkk8ON0rccnl7o9Eas3g00VsMe0sTijniCUThhdRmTUpMONNZvM4bSVi81qjNpwvqz9tjDn1QlyziSrvyQpSaQiJcj1h9mTsfhw-hz+qribzSRCYFDcLDFiU6ZKGRt0ca2REgA)

**Versions**
- TypeScript version: 4.3.5 & 4.4.0-beta
- ts-pattern version: 3.2.4

**Comment by gvergnaud** [comment]
Yeah it's definitely weird! `SeveralAnonymousSelectError` is only supposed to happen if the pattern contains several instances of `select()`, I don't know why it happens in this case. 

Imo the most intuitive behavior would be that `date` remains `any` in this case. 

Thank you for reporting this, I'll try to find the time to have a look as well

**Comment by speigg** [comment]
Also running into this

**Comment by gvergnaud** [comment]
Fixed in v3.3.3
